### PR TITLE
Removing u8 from HTMLIframeElement.rs file

### LINKED ISSUE #11599: Do not use u8 in HTMLIFrameElement::sandbox
The field `sandbox` of `HTMLIFrameElement` should be changed to `Cell<Option<SandboxAllowance>>`.

### PR DESCRIPTION
**Pull request by Coder206**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #11599 (github issue number if applicable).

- [x] These changes do not require tests because @nox did not request any so I am not sure which to do.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12113)

### CODE REVIEW

File: components/script/dom/htmliframeelement.rs [hunk 1]
```diff
@@ -50,15 +50,17 @@ use style::context::ReflowGoal;
...
-    AllowNothing = 0x00,
-    AllowSameOrigin = 0x01,
-    AllowTopNavigation = 0x02,
-    AllowForms = 0x04,
-    AllowScripts = 0x08,
-    AllowPointerLock = 0x10,
-    AllowPopups = 0x20
+bitflags! {
+	#[derive(JSTraceable, HeapSizeOf)]
+    	pub flags SandboxAllowance: u8 {
```
Comment (mbrubeck) [inline components/script/dom/htmliframeelement.rs]: This probably doesn't need to be `pub`.

Comment (Coder206) [inline components/script/dom/htmliframeelement.rs]: @mbrubeck Ok.

File: components/script/dom/htmliframeelement.rs [hunk 2]
```diff
@@ -50,15 +50,17 @@ use style::context::ReflowGoal;
...
-    AllowForms = 0x04,
-    AllowScripts = 0x08,
-    AllowPointerLock = 0x10,
-    AllowPopups = 0x20
+bitflags! {
+    #[derive(JSTraceable, HeapSizeOf)]
+    flags SandboxAllowance: u8 {
+        const ALLOW_NOTHING = 0x00,
+    	const ALLOW_SAME_ORIGIN = 0x01,
+	    const ALLOW_TOP_NAVIGATION = 0x02,
```
Comment (mbrubeck) [inline components/script/dom/htmliframeelement.rs]: nit: Some weird indentation here.

**Comment by mbrubeck** [comment]
r=mbrubeck with the whitespace fixed (see above)

**Comment by Coder206** [comment]
@mbrubeck Thanks :)

**Comment by Coder206** [comment]
@mbrubeck I am not sure why Travic is still giving an error.

**Comment by ConnorGBrewster** [comment]
Convert those tabs to spaces :)
```
Checking files for tidiness...
./components/script/dom/htmliframeelement.rs:54: tab on line
./components/script/dom/htmliframeelement.rs:55: tab on line
./components/script/dom/htmliframeelement.rs:56: tab on line
./components/script/dom/htmliframeelement.rs:57: tab on line
./components/script/dom/htmliframeelement.rs:58: tab on line
./components/script/dom/htmliframeelement.rs:59: tab on line
./components/script/dom/htmliframeelement.rs:60: tab on line
./components/script/dom/htmliframeelement.rs:61: tab on line
./components/script/dom/htmliframeelement.rs:62: tab on line
./components/script/dom/htmliframeelement.rs:63: tab on line
```

**Comment by Coder206** [comment]
@ConnorGBrewster Oh ok, thanks.

**Comment by emilio** [comment]
@bors-servo: r=mbrubeck,emilio

BTW, @Coder206, feel free to run `git config --global user.email <your-email>`, since your commits appear with a `@localhost.localdomain` email. You can read more about that here: https://help.github.com/articles/setting-your-email-in-git/.

Thanks for your contribution! :)

**Comment by Coder206** [comment]
@emilio Thanks :)

**Pull request merged**